File: Implement_Stack_using_Queues_two.cpp
class MyStack {
private:
    std::queue<int> q1;
    std::queue<int> q2;

public:
    MyStack() {}

    void push(int x) {
        q1.push(x);
    }

    int pop() {
        while (q1.size() > 1) {
            q2.push(q1.front());
            q1.pop();
        }

        int poppedVal = q1.front();
        q1.pop();
        std::swap(q1, q2);

        return poppedVal;
    }

    int top() {
        while (q1.size() > 1) {
            q2.push(q1.front());
            q1.pop();
        }

        int topVal = q1.front();
        q2.push(q1.front());
        q1.pop();
        std::swap(q1, q2);

        return topVal;
    }

    bool empty() {
        return q1.empty();
    }
};

File: Implement_Stack_using_Queues_two.java
class MyStack {

    private Queue<Integer> q1;
    private Queue<Integer> q2;

    public MyStack() {
        q1 = new ArrayDeque<>();
        q2 = new ArrayDeque<>();
    }

    public void push(int x) {
        q1.add(x);
    }

    public int pop() {
        while (q1.size() > 1) {
            q2.add(q1.poll());
        }

        int poppedVal = q1.poll();
        Queue<Integer> temp = q1;
        q1 = q2;
        q2 = temp;

        return poppedVal;
    }

    public int top() {
        while (q1.size() > 1) {
            q2.add(q1.poll());
        }

        int topVal = q1.peek();
        q2.add(q1.poll());
        Queue<Integer> temp = q1;
        q1 = q2;
        q2 = temp;

        return topVal;
    }

    public boolean empty() {
        return q1.isEmpty();
    }
}
